config/config.go

```diff
@@ -33,21 +33,24 @@ import (
 )
 
 var (
-	patRulePath         = regexp.MustCompile(`^[^*]*(\*[^/]*)?$`)
-	unchangeableHeaders = map[string]struct{}{
+	patRulePath     = regexp.MustCompile(`^[^*]*(\*[^/]*)?$`)
+	reservedHeaders = map[string]struct{}{
 		// NOTE: authorization is checked specially,
 		// see RemoteWriteConfig.UnmarshalYAML.
 		// "authorization":                  {},
 		"host":                              {},
 		"content-encoding":                  {},
+		"content-length":                    {},
 		"content-type":                      {},
-		"x-prometheus-remote-write-version": {},
 		"user-agent":                        {},
 		"connection":                        {},
 		"keep-alive":                        {},
 		"proxy-authenticate":                {},
 		"proxy-authorization":               {},
 		"www-authenticate":                  {},
+		"accept-encoding":                   {},
+		"x-prometheus-remote-write-version": {},
```

Comment:
I am bit confused, why we have those, never seen them before? For example for remote read we don't version, we negotiate types natively in proto.

Reply:
`x-prometheus-remote-write-version` was something we already had marked as reserved. 

@roidelapluie could you please take a look here? (`accept-encoding` thing).

Reply:
We can't let user override accept-encoding as it would break remote read

Reply:
> x-prometheus-remote-write-version was something we already had marked as reserved. 

Ok, having read reserved if we never use it sounds weird, but not a biggy

Reply:
You do not use it. However, we set it.